### PR TITLE
CMake will not detect a variable as empty string

### PR DESCRIPTION
**Pull request by fran6co**
## Description ##
Testing it with CMake 3.12 it's not detecting that CUDA_TOOLSET is empty

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -210,7 +210,7 @@ if(USE_CUDA)
   find_package(CUDA REQUIRED)
   add_definitions(-DMSHADOW_USE_CUDA=1)
   if(FIRST_CUDA AND (NOT USE_OLDCMAKECUDA))
-    if(CUDA_TOOLSET STREQUAL "")
+    if(NOT CUDA_TOOLSET)
```
Comment (yajiedesign) [inline CMakeLists.txt]: may be ()NOT CUDA_TOOLSET) OR ()CUDA_TOOLSET STREQUAL "") is good

Comment (fran6co) [reply CMakeLists.txt]: NOT checks for empty string and when the variable is not set, so there is no need for the OR

Comment (yajiedesign) [reply CMakeLists.txt]: ok

**Comment by yajiedesign** [comment]
LGTM

**Pull request merged**